org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/ContentModelFormatterParticipant.java

```diff
@@ -61,7 +62,16 @@ public FormatElementCategory getFormatElementCategory(DOMElement element,
 		}
 		Collection<CMDocument> cmDocuments = formattingContext.get(namespaceURI);
 		if (cmDocuments == null) {
-			cmDocuments = contentModelManager.findCMDocument(element);
+			try {
+				cmDocuments = contentModelManager.findCMDocument(element);
+			} catch (CacheResourceDownloadingException e) {
+				// block; we need the content model to format correctly.
+				// if we don't block and instead return null here,
+				// then some parts of the document will be
+				// grammar-aware formatted and others won't be.
+				e.getFuture().join();

```

Comment:
Generaly we should avoid using join which can block the thread pooling.

I think the proper solution is to use CompletableFuture (get cmDocuments) and thenApply (format), but I fear it is a big refactoring.

Perhaps it should be safer to use a timeout (e.getFuture().get(2000) instead of join?